dhall/tests/Dhall/Test/Dhall.hs

```diff
@@ -489,3 +497,83 @@ shouldShowCorrectErrorForInvalidDecoderInUnion =
+        , tCase "tests/config-fixtures/sub/b" "tests/config-fixtures/dhall-config.dhall"
+    ]
+    where
+        tCase :: String -> String -> TestTree
+        tCase startDir expected = testCase ("starting in " ++ startDir) $
+            Dhall.Main.findConfig startDir >>= \case
+                Nothing -> fail "config should always be found (in our fixture tests)"
+                Just result -> do
+                    cwd <- System.Directory.getCurrentDirectory
+                    let rel = System.FilePath.makeRelative cwd result in
+                        assertEqual "the path should be correct" (expected) rel
+
+
+shouldResolveSettings :: TestTree
+shouldResolveSettings =
+        testGroup "Should combine settings from Options (command line) and dhall-config.dhall" [
+                -- (optAcii, optExplain) (confAscii, confExplain)
+            testGroup "config overridden by command line True values" [
+                tCase t t asciiAndExplain
+                , tCase t f asciiAndExplain
+                , tCase t n asciiAndExplain
+            ]
+            , testGroup "config overridden by command line False values" [
+                tCase f t unicodeAndConcise
+                , tCase f f unicodeAndConcise
+                , tCase f n unicodeAndConcise
+            ]
+            , testGroup "command line options not provided, config provided" [
+                tCase n t asciiAndExplain
+                , tCase n f unicodeAndConcise
+            ]
+            , testGroup "neither command line nor config provided" [
+                tCase n n unicodeAndConcise
+            ]
+        ]
+    where
+        -- because ascii and and explain options should be independent, we don't
+        -- need to consider them separately in test fixtures
+        t = (Just True, Just True)
+        f = (Just False, Just False)
+        n = (Nothing, Nothing)
+        asciiAndExplain = (Dhall.Main.ASCII, True)
+        unicodeAndConcise = (Dhall.Main.Unicode, False)
+
+        tCase rawOpts@(optAscii, optExplain) rawConf@(confAscii, confExplain) expected =
+            let msg = "opts: " ++ (show rawOpts) ++ " config: " ++ (show rawConf) in
+            testCase msg $
+                    let opts = toOptions optAscii optExplain in
+                    let confStr = toConfigStr confAscii confExplain in do
+                    result <- Dhall.Main.resolveSettings (Just confStr) opts
+                    assertEqual "resolved settings should be correct" expected result
+
+        toOptions ascii explain = Dhall.Main.Options{
+            ascii = ascii
+            , censor = Dhall.Util.Censor
+            , explain = explain
+            , mode = Dhall.Main.Format{
+                possiblyTransitiveInput = Dhall.Util.NonTransitiveStandardInput
+                , outputMode = Dhall.Util.Write
+            }
+            , plain = False
+        }
+
```

Comment:
I think it would be more maintainable here to use Dhall's built-in ability to turn Haskell values into Dhall values, but I couldn't figure out how to do it.